src/server/middleware/error.middleware.ts

```diff
@@ -0,0 +1,21 @@
+import { Request, Response, NextFunction } from 'express';
+import { CustomError, CustomErrors, errorTypeToStatusCode } from '../custom-error.model';
+
+export const isCustomError = (error: Error | CustomError) => error instanceof CustomError;
+
+export const errorMiddleware = (
+  error: Error | CustomError,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (!isCustomError(error)) {
+    error = new CustomError({
```

Comment:
yeah I had this feeling - i was reading a blogpost where a guy did something similar so I rolled with it. I thought bc it was only being used by my middleware it shouldn't have many problems. the gist was is that if it wasn't an `AuthkunError` I wanted to make it one, so it would automatically handle the status code etc